src/community/taskmanager/src/main/java/org/geoserver/taskmanager/report/impl/SimpleReportBuilderImpl.java

```diff
@@ -0,0 +1,66 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.taskmanager.report.impl;
+
+import org.geoserver.taskmanager.data.BatchElement;
+import org.geoserver.taskmanager.data.BatchRun;
+import org.geoserver.taskmanager.data.Run;
+import org.geoserver.taskmanager.data.Task;
+import org.geoserver.taskmanager.report.Report;
+import org.geoserver.taskmanager.report.ReportBuilder;
+import org.geoserver.taskmanager.report.Report.Type;
+import org.springframework.stereotype.Service;
+
+/**
+ * A very simple report builder.
+ * Contains all information that matters in simple text format.
+ * 
+ * @author Niels Charlier
+ *
+ */
+@Service
+public class SimpleReportBuilderImpl implements ReportBuilder {
+
```

Comment:
Consider adding logging for failure cases here?

Reply:
not sure what you mean... exceptions and batch start/stop/cancel events are sent to the general log in BatchJobImpl . If you want to send information about batch results elsewhere you can write your own reportservice

Reply:
OK. I thought it might be useful to provide some additional logging context information here, but it was a "maybe" level suggestion.